docs/calendar.md

```diff
@@ -347,6 +347,38 @@ Temporal.Calendar.from('iso8601').fields(['month', 'day']);
 ```
 <!-- prettier-ignore-end -->
 
+### calendar.**mergeFields**(_fields_: object, _extraFields_: object) : object
+
+**Parameters:**
+
+- `fields` (object): An object with properties representing calendar units.
+- `extraFields` (object): Another object with properties representing calendar units.
+
+**Returns:** an object with properties from both `fields` and `extraFields`.
+
+This method does not need to be called directly except in specialized code.
+It is called indirectly when using the `with()` methods of `Temporal.PlainDateTime`, `Temporal.PlainDate`, `Temporal.PlainMonthDay`, `Temporal.PlainYearMonth`, and `Temporal.ZonedDateTime`.
+
+Custom calendars should override this method if they allow a calendar unit to be specified in more than one way.
+(For example, the Gregorian calendar allows years to be specified either by a `year` property or a combination of `era` and `eraYear`.)
+
+When subclassing `Temporal.Calendar`, this method doesn't need to be overridden, unless your calendar allows this with other properties than the built-in calendars do.
```

Comment:
I'm a bit hesitant to spend a lot of space on an explanation, because I think very few people are ever seriously going to write userland code that does this, but I have rephrased it a bit. :smile: